xds/src/main/java/io/grpc/xds/EnvoyServerProtoData.java

```diff
@@ -248,21 +310,39 @@ public boolean equals(Object o) {
       }
       FilterChainMatch that = (FilterChainMatch) o;
       return destinationPort == that.destinationPort
-          && java.util.Objects.equals(prefixRanges, that.prefixRanges)
-          && java.util.Objects.equals(applicationProtocols, that.applicationProtocols);
+          && Objects.equals(prefixRanges, that.prefixRanges)
+          && Objects.equals(applicationProtocols, that.applicationProtocols)
+          && Objects.equals(sourcePrefixRanges, that.sourcePrefixRanges)
+          && sourceType == that.sourceType
+          && Objects.equals(sourcePorts, that.sourcePorts);
     }
 
     @Override
     public int hashCode() {
-      return java.util.Objects.hash(destinationPort, prefixRanges, applicationProtocols);
+      return Objects.hash(
+          destinationPort,
+          prefixRanges,
+          applicationProtocols,
+          sourcePrefixRanges,
+          sourceType,
+          sourcePorts);
     }
 
     @Override
     public String toString() {
       return "FilterChainMatch{"
-          + "destinationPort=" + destinationPort
-          + ", prefixRanges=" + prefixRanges
-          + ", applicationProtocols=" + applicationProtocols
+          + "destinationPort="
+          + destinationPort
+          + ", prefixRanges="
+          + prefixRanges
+          + ", applicationProtocols="
+          + applicationProtocols
+          + ", sourcePrefixRanges="
+          + sourcePrefixRanges
+          + ", sourceType="
+          + sourceType
+          + ", sourcePorts="
+          + sourcePorts
```

Comment:
nit: you may use `MoreObjects.toStringHelper()` as in `EnvoyProtoData`

Reply:
Thx for the tip. This was generated by the IDE so I just used that option

Reply:
There is also guava option to generate toString by the IDE.

Reply:
Yes, that's what I ended up doing.